src/main/kotlin/it/pagopa/wallet/scheduler/scheduledjob/PaymentWalletScheduledJob.kt

```diff
@@ -0,0 +1,46 @@
+package it.pagopa.wallet.scheduler.scheduledjob
+
+import it.pagopa.wallet.scheduler.config.properties.PaymentWalletJobConfiguration
+import it.pagopa.wallet.scheduler.jobs.config.OnboardedPaymentWalletJobConfiguration
+import it.pagopa.wallet.scheduler.jobs.paymentwallet.OnboardedPaymentWalletJob
+import java.time.Duration
+import java.time.Instant
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Service
+
+@Service
+class PaymentWalletScheduledJob(
+    @Autowired private val onboardedPaymentWalletJob: OnboardedPaymentWalletJob,
+    @Autowired private val paymentWalletJobConfiguration: PaymentWalletJobConfiguration
+) {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    @Scheduled(cron = "\${paymentWalletJob.execution.cron}")
+    fun processOnboardedPaymentWallets() {
+        val startTime = Instant.now()
+        onboardedPaymentWalletJob
+            .process(
+                OnboardedPaymentWalletJobConfiguration(
+                    startDate = paymentWalletJobConfiguration.startDate,
+                    endDate = paymentWalletJobConfiguration.endDate
+                )
+            )
+            .subscribe(
+                {
+                    logger.info(
+                        "Payment wallets processed successfully. Latest created date processed wallet: [{}]",
+                        it
+                    )
+                },
+                { logger.error("Exception processing payment wallets", it) },
```

Comment:
Avoid logging whole exception to limit risk of stacktrace being exposed
 
                `{ error ->
                    logger.error("Exception processing payment wallets: {}", error.message)
                },`

Reply:
Why we should not log stack trace?

Why do you talk about expose stack trace here? those are application logs, stack trace is not put into any response body here.